### PR TITLE
feat: edit life events

### PR DESCRIPTION
**Pull request by asbiin**

### CODE REVIEW

File: app/Domains/Contact/ManageLifeEvents/Services/EditLifeEvent.php [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    /**
+     * Create a life event.
+     */
+    public function execute(array $data): LifeEvent
+    {
+        $this->data = $data;
+        $this->validate();
+        $this->store();
```
Comment (djaiss) [inline app/Domains/Contact/ManageLifeEvents/Services/EditLifeEvent.php]: technically, this should be `update` and not `store`, if we want to be picky.

File: app/Domains/Contact/ManageLifeEvents/Services/EditLifeEvent.php [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+use App\Interfaces\ServiceInterface;
+use App\Models\Contact;
+use App\Models\LifeEvent;
+use App\Models\LifeEventType;
+use App\Models\TimelineEvent;
+use App\Services\BaseService;
+use Carbon\Carbon;
+use Illuminate\Support\Collection;
+
+class EditLifeEvent extends BaseService implements ServiceInterface
```
Comment (djaiss) [inline app/Domains/Contact/ManageLifeEvents/Services/EditLifeEvent.php]: All the services in the app that edit something are called `UpdateXXX`, not `EditXXX`. Therefore, this service should be named `UpdateLifeEvent`.

**Review by djaiss — COMMENTED** [review]
Bravo

File: resources/js/Shared/Modules/LifeEvent.vue [hunk 3]
```diff
@@ -75,6 +75,12 @@ const destroyLifeEvent = (timelineEvent, lifeEvent) => {
   }
 };
 
+const refreshLifeEvent = (timelineEvent, lifeEvent) => {
+  var id = localTimelines.value.findIndex((x) => x.id === timelineEvent.id);
+  var lifeEventId = localTimelines.value[id].life_events.findIndex((x) => x.id === lifeEvent.id);
```
Comment (asbiin) [inline resources/js/Shared/Modules/LifeEvent.vue]:
```suggestion
  let id = localTimelines.value.findIndex((x) => x.id === timelineEvent.id);
  let lifeEventId = localTimelines.value[id].life_events.findIndex((x) => x.id === lifeEvent.id);
```

**Pull request merged**